src/runtime/internal/style_manager.ts

```diff
@@ -53,14 +59,19 @@ export function delete_rule(node: Element & ElementCSSInlineStyle, name?: string
 		)
 		.join(', ');
 
-	if (name && !--active) clear_rules();
+	active = Math.max(0, active - 1);
+	if (name && !active) clear_rules();
 }
 
 export function clear_rules() {
 	raf(() => {
 		if (active) return;
-		let i = stylesheet.cssRules.length;
-		while (i--) stylesheet.deleteRule(i);
-		current_rules = {};
+		activeDocs.forEach(doc => {
+			const [ stylesheet ] = docStyles.get(doc);
+			let i = stylesheet.cssRules.length;
+			while (i--) stylesheet.deleteRule(i);
+			docStyles.set(doc, [ stylesheet, {} ]);
+		});
+		activeDocs.clear();
```

Comment:
oh! we can *definitely* use a `Map` instead of a `WeakMap` if we're clearing `activeDocs` (can we change it to `active_docs`?)